ansible/library/load_extra_dpu_config.py

```diff
@@ -0,0 +1,107 @@
+            argument_spec=dict(
+                hwsku=dict(type='str', required=True),
+                host_username=dict(type='str', required=False, default='admin'),
+                host_password=dict(type='str', required=False, default='password', no_log=True)
+            ),
+            supports_check_mode=False
+        )
+
+        self.hwsku = self.module.params['hwsku']
+        self.host_username = self.module.params['host_username']
+        self.host_password = self.module.params['host_password']
+
+        try:
+            self.hwsku_config = smartswitch_hwsku_config[self.hwsku]
+            self.dpu_num = self.hwsku_config.get('dpu_num', 0)
+
+            if self.dpu_num == 0:
+                self.module.fail_json(msg="No DPUs defined for hwsku: {}".format(self.hwsku))
+        except KeyError:
+            self.module.fail_json(msg="No DPU configuration found for hwsku: {}".format(self.hwsku))
+
+    def connect_to_dpu(self, dpu_ip):
+        """Establish an SSH connection to the DPU"""
+        try:
+            ssh = paramiko.SSHClient()
+            ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            ssh.connect(dpu_ip, username=self.host_username, password=self.host_password, timeout=30)
+            return ssh
+        except Exception as e:
+            self.module.fail_json(msg="Failed to connect to DPU {}: {}".format(dpu_ip, str(e)))
+            return None
+
+    def transfer_to_dpu(self, ssh, dpu_ip):
+        """Transfer the configuration file to the DPU"""
+        try:
+            with ssh.open_sftp() as sftp:
+                sftp.put(SRC_DPU_CONFIG_FILE, DST_DPU_CONFIG_FILE)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to transfer file to DPU {}: {}".format(dpu_ip, str(e)))
+
+    def execute_command(self, ssh, dpu_ip, command):
+        """Execute a command on the DPU"""
+        _, stdout, stderr = ssh.exec_command(command)
+        exit_code = stdout.channel.recv_exit_status()
+        if exit_code != 0:
+            self.module.fail_json(msg="Command failed on DPU {} with exit status {}: {}".format(
+                dpu_ip, exit_code, stderr.read().decode('utf-8')))
+
+    def configure_dpus(self):
+        """Configure all DPUs based on the hardware SKU configuration"""
+        if not os.path.isfile(SRC_DPU_CONFIG_FILE):
+            self.module.fail_json(msg="DPU config file not found: {}".format(SRC_DPU_CONFIG_FILE))
+
+        for i in range(0, self.dpu_num):
+            dpu_ip = DPU_HOST_IP_BASE.format(i + 1)
+
+            ssh = self.connect_to_dpu(dpu_ip)
+            if not ssh:
+                self.module.fail_json(msg="Failed to ssh to DPU: {}".format(dpu_ip))
+
+            try:
+                self.transfer_to_dpu(ssh, dpu_ip)
+                self.execute_command(ssh, dpu_ip, GEN_FULL_CONFIG_CMD)
+                self.execute_command(ssh, dpu_ip, CONFIG_RELOAD_CMD)
+                self.execute_command(ssh, dpu_ip, CONFIG_SAVE_CMD)
```

Comment:
Why config save after reload? will the config file content change by reload from it and then save to it?

Reply:
CONFIG_SAVE_CMD is to save to default /etc/sonic/config_db.json.
Delete config is to delete transfered file such as /tmp/dpu_extra.json